src/core/core.js

```diff
@@ -345,6 +346,7 @@ export default class Core {
         location: this._getLocationPayload(),
         skipSpellCheck: this.storage.get(StorageKeys.SKIP_SPELL_CHECK),
         queryTrigger: queryTriggerForApi,
+        sessionId: this.getSessionId(),
```

Comment:
One thing the spec is a little unclear on is which requests we need to include `sessionId` with. Is it only the search requests or on autocomplete as well? Rose may need to weigh in there.

Reply:
The browser handles cookies in the case of session storage failing. 

When sessionTrackingEnabled is false, getSessionId() returns null. When that happens, in the browser, I don't see a sessionId query parameter which indicates to me that setting it to null results in it being filtered out. 

I assumed we would only use search requests because a search request is called whenever we refresh the page. Since a search request occurs when the page loads for the first time and we generate a sessionId then. It might be redundant to include it in the autocomplete as well since we don't update our sessionId if a sessionId already exists. So for example, the page loads and a search query happens and we generate our first sessionID. another search happens and our sessionID doesn't change because we already have an existing one.

Reply:
For tracking purposes, they may want to associate various auto-complete calls or question submission calls with a particular `sessionId`. Here, I think it's a question of what the back-end expects. If we don't include a `sessionId` on an auto-complete request, I'm not sure the back-end has any way of knowing which session to associate it with. @mdavish, do you know what the back-end expects here? Should all requests made by the SDK (excluding Analytics) have a `sessionId`?

Reply:
Okay, heard back from Product. We only want `sessionId` for the searches!